projects/transformers/export_finetuning_results.py

```diff
@@ -184,19 +188,98 @@ def get_metric_from_runs(self, task, metric):
             metric_all_runs[2] could return eval_loss at every time step
             on the 3rd run for a given task.
         """
+        metrics = [m for m in self[task].reporting_metrics]
+
+        if hasattr(self[task], "score_mats"):
+            return self[task].score_mats
+
         n_steps = len(self[task].all_results[0]["steps"])
         n_runs = len(self[task].all_results)
 
-        metric_all_runs = np.zeros((n_runs, n_steps))
-        for run_idx in range(n_runs):
-            metric_all_runs = np.array(
-                self[task].all_results[run_idx][metric])
+        metrics_all_runs = []
+        for _ in range(len(self[task].reporting_metrics)):
+            metrics_all_runs.append(np.zeros((n_runs, n_steps)))
+
+        for m in range(len(metrics)):
+            for run_idx in range(n_runs):
+                metrics_all_runs[m][run_idx, :] = np.array(
+                    self[task].all_results[run_idx][metrics[m]])
 
-        return metric_all_runs
+        self[task].score_mats = metrics_all_runs
+
+        return metrics_all_runs
+
+    def get_best_scores_per_run(self, task):
+        metrics = self[task].reporting_metrics
+        metrics_all_runs = self.get_metrics_from_runs(task)
+        if len(metrics) > 1:
+            metric_all_runs = np.zeros_like(metrics_all_runs[0])
+            for m in range(len(metrics)):
+                metric_all_runs += metrics_all_runs[m]
+            metric_all_runs /= len(metrics)
+        else:
+            metric_all_runs = metrics_all_runs[0]
+        best_idx_per_run = np.argmax(metric_all_runs, axis=1)
+        best_avg_metric_scores = metric_all_runs[np.arange(len(self[task])),
+                                                 best_idx_per_run]
+
+        best_metric_scores = []
+        for m in range(len(metrics)):
+            data = metrics_all_runs[m][np.arange(len(self[task])), best_idx_per_run]
+            best_metric_scores.append(data)
+        return best_avg_metric_scores, best_metric_scores
+
+    # TODO
```

Comment:
We try not to leave commented code, it is not good style - if it is deprecated, can be a function with a deprecation notice in the docstring that we eventually remove once we deem it will no longer be required.